lib/api/bucketPutACL.js

```diff
@@ -281,12 +285,15 @@ export default function bucketPutACL(authInfo, request, log, callback) {
         if (err) {
             log.trace('error processing request', { error: err,
                 method: 'bucketPutACL' });
-            return callback(err);
+            if (err === errors.NoSuchBucket) {
+                return callback(err);
+            }
```

Comment:
I assume there is a reason you don't want `corsHeaders` when the error is NoSuchBucket. 

You might then just return the callback here with `corsHeaders` as the second argument rather than using an else-statement for the `pushMetric` call.